packages/shared/src/components/modals/OnboardingModal.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {
+  KeyboardEvent,
+  MouseEvent,
+  ReactElement,
+  ReactNode,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import { ModalProps } from './StyledModal';
+import SteppedModal from './SteppedModal';
+import ThemeOnboarding from '../onboarding/ThemeOnboarding';
+import FilterOnboarding from '../onboarding/FilterOnboarding';
+import LayoutOnboarding from '../onboarding/LayoutOnboarding';
+import IntroductionOnboarding from '../onboarding/IntroductionOnboarding';
+import FeaturesContext from '../../contexts/FeaturesContext';
+import { OnboardingStep } from '../onboarding/common';
+import { LoginTrigger } from '../../lib/analytics';
+
+const INTRODUCTION_ADDITIONAL_STEP = 1;
+
+interface OnboardingModalProps extends ModalProps {
+  trigger?: string;
+  onRegistrationSuccess?: () => void;
+}
+
+const backCopy = ['Skip', 'Close', 'Back', 'Back'];
```

Comment:
How does this work? What happens when AB decided to only show one screen?

Reply:
This should still be as what is intended. First is skip and Close on start of actual onboarding.

Reply:
Not quite sure I understand the purpose then?
Is it not assigned to each step?

Perhaps we can turn this into a dictionary?

Reply:
I think we will have to keep it an array, the index represents which step it will be reflected in.

The introduction will have index 0 which is the skip and so on.

The footer is abstracted inside the component but we have control over its labels and callbacks if any.